lib/elsa/util.ex

```diff
@@ -105,8 +105,8 @@ defmodule Elsa.Util do
   Return the number of partitions for a given topic. Bypasses the need for a persistent client
   for lighter weight interactions from one-off calls.
   """
-  @spec partition_count(keyword(), String.t()) :: integer()
-  def partition_count(endpoints, topic) do
+  @spec partition_count(keyword | Elsa.connection() | pid, String.t()) :: integer()
```

Comment:
why are you handling a single topic here when the first arguments is a keyword list of endpoints, but the one below when the first argument is an atom or a pid looks to take a list of topics?

Reply:
line 110 vs 120

Reply:
:brod.get_metadata take a list of topics, :brod_client.get_metadata only takes a single topic.

Reply:
Why the two different strategies?

Reply:
the new one uses an existing connection rather then establishing a new connection to just get metadata.

Reply:
Got it; didn't realize the other one would require a full extra roundtrip to the cluster. Assuming that isn't essentially required to ensure up-to-date accuracy (the existing connection won't have a potentially inaccurate cache of info) that makes sense to me.